app/code/core/Mage/Catalog/Model/Product.php

```diff
@@ -579,25 +579,23 @@ public function getCategory()
     }
 
     /**
-     * Set assigned category IDs array to product
+     * Replace product's category ID(s)
```

Comment:
I'd keep it as it is.

Reply:
`Set assigned`, to me, does not clearly state that existing data will be removed. And the input it not always an array.

Reply:
IMHO it is clear. "set" vs "add"? Finally you set an array of category IDs ...

I'd keep the changes as small as possible, and i dont see any benefit here.